### DISCUSSION TITLE
OpenIDC Support for Opengrok

### DISCUSSION THREAD

**Discussion by Leo-Adlakha** [oracle/opengrok#4045] [Q&A]
Hi, I wanted to have OpenIDC authentication with Opengrok, does Opengrok support that ?? If it does, then is there any documentation I can refer from where I can configure the OpenIDC with Opengrok.

**Comment by vladak** [comment]
Assuming you are eventually after authorization. This looks to be similar in nature to https://github.com/oracle/opengrok/discussions/4046. Basically, as long as the authentication piece can supply information about the authenticated user (either as part of the request structure or via HTTP headers) to the OpenGrok webapp, a `UserPlugin` decoder can be written to extract the data and pass it to the authorization stack.
